### PR TITLE
sqrtf.c OK

### PR DESCRIPTION
**Pull request by EllipticEllipsis**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

I must get out of the habit of doing such large PRs...

### CODE REVIEW

File: src/boot_O2/sqrtf.c [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#include "global.h"
+
+#ifndef __GNUC__
+#define __builtin_sqrtf sqrtf
```
Comment (AngheloAlf) [inline src/boot_O2/sqrtf.c:4]: If __GNUC__ is not defined would this produce an infinite recursion, right?

Comment (EllipticEllipsis) [reply src/boot_O2/sqrtf.c]: One would think so, but 
```c
f32 sqrtf(f32 f) {
    return sqrtf(f);
}
```
compiles fine, with no warnings and gives OK.

Comment (AngheloAlf) [reply src/boot_O2/sqrtf.c]: Really? I need an explanation...

Comment (Thar0) [reply src/boot_O2/sqrtf.c]: `sqrtf` is a compiler intrinsic, meaning when it's used inside a function it doesn't actually call the function `sqrtf`, a `sqrt.s` MIPS instruction gets inlined instead, so there is no recursion happening here despite how it appears.

Comment (EllipticEllipsis) [reply src/boot_O2/sqrtf.c]: Does that mean that this function (the one defined in this file) is actually pointless, then? Other functions will never call it, instead using the intrinsic?

Comment (AngheloAlf) [reply src/boot_O2/sqrtf.c]: That means that this function is never used? Every call to this function would trigger that compiler intrinsic, right?

**Review by AngheloAlf — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Comment by Kenix3** [comment]
Needs +1 reviewer for merge.

**Review by isghj5 — APPROVED** [review]

**Pull request merged**